go-controller/pkg/ovn/gateway_init.go

```diff
@@ -592,34 +600,60 @@ func addPolicyBasedRoutes(nodeName, mgmtPortIP string, hostIfAddr *net.IPNet) er
 // 		a1b876f6-5ed4-4f88-b09c-7b4beed3b75f,ip4.src == 10.244.1.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-control-plane */,169.254.0.1
 // 		0f5af297-74c8-4551-b10e-afe3b74bb000,ip4.src == 10.244.0.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-worker2 */,169.254.0.1
 
-// To check if the management port IP has changed for the node that concerns it,
-// and updates that route in case it found a matching one.
+// The function checks to see if the mgmtPort IP has changed, or if match criteria has changed
+// and removes stale policies for a node. It also adds new policies for a node at a specific priority.
 // This is ugly (since the node is encoded as a comment in the match),
 // but a necessary evil as any change to this would break upgrades and
 // possible downgrades. We could make sure any upgrade encodes the node in
 // the external_id, but since ovn-kubernetes isn't versioned, we won't ever
 // know which version someone is running of this and when the switch to version
 // N+2 is fully made.
-func syncPolicyBasedRoutes(nodeName, mgmtPortIP, match, priority, nexthops, compareTo string) error {
-	found := false
-	policiesCompare, err := findPolicyBasedRoutes(priority, compareTo)
+func syncPolicyBasedRoutes(nodeName, mgmtPortIP string, matches []string, priority, nexthop string) error {
+	policiesCompare, err := findPolicyBasedRoutes(priority, "match,nexthops")
 	if err != nil {
 		return fmt.Errorf("unable to list policies, err: %v", err)
 	}
+
+	// create a map to track matches found
+	matchTracker := sets.NewString(matches...)
```

Comment:
Ah, that might fix what I mentioned about `otherHostAddrs`. But I am not sure I like it